### PR TITLE
 Everywhere: Use Skia painter by default

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**
Skia painter is visibly faster than LibGfx painter and has more complete
CSS transforms support. With this change:
- On Linux, it will try to use Vulkan-backend with fallback to
  CPU-backend
- On macOS it will try to use Metal-backend with fallback to
  CPU-backend
- headless-browser always runs with CPU-backend in layout mode

### CODE REVIEW

File: Tests/LibWeb/Screenshot/images/css-gradients-ref.png
Comment (jamierocks) [inline Tests/LibWeb/Screenshot/images/css-gradients-ref.png:1]: The last two in the file appear worse under Skia.

image

Comment (kalenikaliaksandr) [reply Tests/LibWeb/Screenshot/images/css-gradients-ref.png:1]: yes, Skia doesn't use antialiasing for radial gradients. I think that's probably acceptable for now. we could investigate if that's possible to fix in the future.

**Comment by kalenikaliaksandr** [comment]
ops, forgot to update SVGMaskable::calculate_mask_of_svg() to use Skia painter. converting to a draft for now.

File: Tests/LibWeb/Screenshot/images/css-gradients-ref.png
Comment (jamierocks) [reply Tests/LibWeb/Screenshot/images/css-gradients-ref.png:1]: I presume that neither use Skia - but it has anti-aliasing on Firefox (left), Gnome Web (right).

image

Also appears that Chromium has anti-aliasing.

image

Comment (kalenikaliaksandr) [reply Tests/LibWeb/Screenshot/images/css-gradients-ref.png:1]: sure, I bet there is a way to make Skia use antialiasing for radial gradients. let's open an issue for that. still, that does not seem like terrible enough problem to block us from switching default.

**Comment by circl-lastname** [comment]
https://github.com/LadybirdBrowser/ladybird/blob/0384b484bd61b12c0ab6f96ffbf7d4015760b7ab/Userland/Libraries/LibWeb/Painting/PaintableBox.cpp#L1016-L1032

Gfx::Painter is used in here, will this change also make this code obsolete? That would leave the 2D Canvas Context as the only remaining user of Gfx::Painter

**Comment by kalenikaliaksandr** [comment]
> https://github.com/LadybirdBrowser/ladybird/blob/0384b484bd61b12c0ab6f96ffbf7d4015760b7ab/Userland/Libraries/LibWeb/Painting/PaintableBox.cpp#L1016-L1032
> 
> Gfx::Painter is used in here, will this change also make this code obsolete? That would leave the 2D Canvas Context as the only remaining user of Gfx::Painter

yes, eventually mask painting will be using Skia too, but currently there is no Skia API wrapper that would be suitable to use here as a replacement.

**Pull request merged**